pgmpy/inference/base.py

```diff
@@ -80,3 +80,40 @@ def __init__(self, model):
             for factor in model.get_factors():
                 for var in factor.variables:
                     self.factors[var].append(factor)
+
+
+class ApproxInference:
+    """
+    Base class for the approximate inference algorithm.
+    Parameters
+    ----------
+    model: pgmpy.models.MarkovModel
+    We need a pairwise MarkovModel for Approx. inference.
+    """
+    def __init__(self, model):
+
+        # List of the variables in the present pairwise Markov Model
+        self.variables = []
```

Comment:
Instead of creating another new class, use the ```Inference``` class. It has all the variables that you require.